### ISSUE TITLE
There is a difference on the code coverage result format 

### ISSUE THREAD

**Issue by fedegiust** [forcedotcom/cli#3077]
### Summary
There is an inconsistency for running tests between deployment and running them directly.
When running a deployment there is an option for code coverage formatters which give a number of options, which is great! I like the html format.

![image](https://github.com/user-attachments/assets/8eb165c1-abb6-4f21-a811-a10f8d6550ee)

But when running apex run test the options are limited

![image](https://github.com/user-attachments/assets/ebb1d59f-7d80-4ba1-b345-ee74a7b26659)

![image](https://github.com/user-attachments/assets/713efcd0-9e82-4e56-a0d1-1a8cbb6efe79)

### Steps To Reproduce

```sf apex run test --test-level RunLocalTests --output-dir temp --synchronous --result-format html --code-coverage```

**Comment by fedegiust** [comment]
In this case it happens across all versions

```
{
  "architecture": "win32-x64",
  "cliVersion": "@salesforce/cli/2.59.6",
  "nodeVersion": "node-v20.17.0",
  "osVersion": "Windows_NT 10.0.19045",
  "rootPath": "C:\\Users\\Usuario\\AppData\\Local\\sf\\client\\2.59.6-767cea4",
  "shell": "C:\\Program Files\\Git\\usr\\bin\\bash.exe",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.2.3 (core)",
    "@oclif/plugin-commands 4.0.14 (core)",
    "@oclif/plugin-help 6.2.11 (core)",
    "@oclif/plugin-not-found 3.2.20 (core)",
    "@oclif/plugin-plugins 5.4.7 (core)",
    "@oclif/plugin-search 1.2.8 (core)",
    "@oclif/plugin-update 4.5.6 (core)",
    "@oclif/plugin-version 2.2.12 (core)",
    "@oclif/plugin-warn-if-update-available 3.1.15 (core)",
    "@oclif/plugin-which 3.2.13 (core)",
    "@salesforce/cli 2.59.6 (core)",
    "apex 3.4.9 (core)",
    "api 1.2.1 (core)",
    "auth 3.6.56 (core)",
    "data 3.6.5 (core)",
    "deploy-retrieve 3.12.7 (core)",
    "info 3.4.6 (core)",
    "limits 3.3.30 (core)",
    "marketplace 1.2.25 (core)",
    "org 4.5.8 (core)",
    "packaging 2.8.4 (core)",
    "schema 3.3.29 (core)",
    "settings 2.3.19 (core)",
    "signups 2.5.14 (user) published 45 days ago (Sat Sep 07 2024) (latest is 2.5.23)",
    "sobject 1.4.35 (core)",
    "source 3.5.19 (core)",
    "telemetry 3.6.12 (core)",
    "templates 56.3.18 (core)",
    "trust 3.7.27 (core)",
    "user 3.5.29 (core)",
    "@salesforce/sfdx-scanner 4.6.0 (user) published 28 days ago (Tue Sep 24 2024)",
    "sfdmu 4.36.3 (user) published 32 days ago (Thu Sep 19 2024)",
    "sfdx-git-delta 5.45.0 (user) published 32 days ago (Fri Sep 20 2024) (latest is 5.49.0)",
    "sfdx-plugin-source-read 1.2.0 (user) published 364 days ago (Tue Oct 24 2023)"
  ]
}
```

**Comment by iowillhoit** [comment]
Hey @fedegiust, thanks for sharing this! I am going to convert this into a Github Discussion since it is more of a feature request than a bug report. We have a few other things outline for a plugin-apex revamp that we will _hopefully_ get prioritized for next year. 

In the meantime as a work-around, you could use the `--dry-run` flag on `sf project deploy start` to get your `html` code coverage output without actually deploying your source
